### PR TITLE
Add a new log method to span that allows for setting the time

### PR DESCRIPTION
**Pull request by devinsba**
I have a use case where we have the time where an event took place but not the ability to create the log as it happens. This method will allow us to still log those events to sleuth.

This also makes the OpenTracing adapter I am working on more feature complete so that is good too.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@devinsba Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=501)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=501) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@devinsba Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=501)!

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Span.java [hunk 1]
```diff
@@ -344,6 +344,13 @@ public void logEvent(String event) {
 	}
 
 	/**
+	 * Add a {@link Log#event event} to a specific point in the timeline associated with this span.
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Span.java]: probably makes sense to indicate the unit of the timestamp

**Comment by marcingrzejszczak** [comment]
Congrats on the PR @devinsba :D

**Pull request merged**